src/main/java/appeng/core/worlddata/SpawnData.java

```diff
@@ -0,0 +1,222 @@
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.util.Collection;
+import java.util.LinkedList;
+import javax.annotation.Nonnull;
+
+import com.google.common.base.Preconditions;
+
+import net.minecraft.nbt.CompressedStreamTools;
+import net.minecraft.nbt.NBTTagCompound;
+
+import appeng.core.AELog;
+
+
+/**
+ * @author thatsIch
+ * @version rv3 - 30.05.2015
+ * @since rv3 30.05.2015
+ */
+final class SpawnData implements IWorldSpawnData, IOnWorldStartable, IOnWorldStoppable
+{
+	@Nonnull
+	private final File spawnDirectory;
+
+	public SpawnData( @Nonnull final File spawnDirectory )
+	{
+		Preconditions.checkNotNull( spawnDirectory );
+		Preconditions.checkArgument( spawnDirectory.isDirectory() );
+
+		this.spawnDirectory = spawnDirectory;
+	}
+
+	@Override
+	public void onWorldStart()
+	{
+
+	}
+
+	@Override
+	public void setGenerated( int dim, int chunkX, int chunkZ ) {
+		synchronized( SpawnData.class )
+		{
+			NBTTagCompound data = this.loadSpawnData( dim, chunkX, chunkZ );
+
+			// edit.
+			data.setBoolean( chunkX + "," + chunkZ, true );
+
+			this.writeSpawnData( dim, chunkX, chunkZ, data );
+		}
+	}
+
+	@Override
+	public boolean hasGenerated( int dim, int chunkX, int chunkZ )
+	{
+		synchronized( SpawnData.class )
+		{
+			NBTTagCompound data = this.loadSpawnData( dim, chunkX, chunkZ );
+			return data.getBoolean( chunkX + "," + chunkZ );
+		}
+	}
+
+	@Override
+	public boolean addNearByMeteorites( int dim, int chunkX, int chunkZ, NBTTagCompound newData )
+	{
+		synchronized( SpawnData.class )
+		{
+			NBTTagCompound data = this.loadSpawnData( dim, chunkX, chunkZ );
+
+			// edit.
+			int size = data.getInteger( "num" );
+			data.setTag( String.valueOf( size ), newData );
+			data.setInteger( "num", size + 1 );
+
+			this.writeSpawnData( dim, chunkX, chunkZ, data );
+
+			return true;
+		}
+	}
+
+	@Override
+	public Collection<NBTTagCompound> getNearByMeteorites( int dim, int chunkX, int chunkZ )
+	{
+		Collection<NBTTagCompound> ll = new LinkedList<NBTTagCompound>();
+
+		synchronized( SpawnData.class )
+		{
+			for( int x = -1; x <= 1; x++ )
+			{
+				for( int z = -1; z <= 1; z++ )
+				{
+					int cx = x + ( chunkX >> 4 );
+					int cz = z + ( chunkZ >> 4 );
+
+					NBTTagCompound data = this.loadSpawnData( dim, cx << 4, cz << 4 );
+
+					if( data != null )
+					{
+						// edit.
+						int size = data.getInteger( "num" );
+						for( int s = 0; s < size; s++ )
+						{
+							ll.add( data.getCompoundTag( String.valueOf( s ) ) );
+						}
+					}
+				}
+			}
+		}
+
+		return ll;
+	}
+
+	@Override
+	public void onWorldStop()
+	{
+
+	}
+
+	private NBTTagCompound loadSpawnData( int dim, int chunkX, int chunkZ )
+	{
+		if( !Thread.holdsLock( SpawnData.class ) )
+		{
+			throw new IllegalStateException( "Invalid Request" );
+		}
+
+		NBTTagCompound data = null;
+		File file = new File( this.spawnDirectory, dim + "_" + ( chunkX >> 4 ) + '_' + ( chunkZ >> 4 ) + ".dat" );
```

Comment:
It might be an idea to move the string generation into its own method. So at least we can test the generated string and guarantee it will always stay consistent.